### PR TITLE
Remove END_TAG

### PR DESCRIPTION
**Pull request by fdplacido**
BEGINRELEASENOTES
- !! Breaking change!!
- Remove the need for an END_TAG in the options file: replace it with a map/dict of key-values
- Adapt XML to Python converter to fit for the change
- Adapt tests with new format

ENDRELEASENOTES

### CODE REVIEW

File: k4MarlinWrapper/src/components/MarlinProcessorWrapper.cpp [hunk 1]
```diff
@@ -104,48 +104,34 @@ StatusCode MarlinProcessorWrapper::loadProcessorLibraries() const {
...
 
 std::shared_ptr<marlin::StringParameters> MarlinProcessorWrapper::parseParameters(
-  const Gaudi::Property<std::vector<std::string>>& parameters,
+  const Gaudi::Property<std::map<std::string, std::vector<std::string>>>& parameters,
   std::string& verbosity) const
 {
   auto parameters_ptr = std::make_shared<marlin::StringParameters>();
   info() << "Parameter values for: " << name() << " of type " << std::string(m_processorType) << endmsg;
-  std::string              parameterName   = "";
   std::vector<std::string> parameterValues = {};
```
Comment (andresailer) [inline k4MarlinWrapper/src/components/MarlinProcessorWrapper.cpp:112]: this can be moved inside the loop over parameters now.

Comment (fdplacido) [reply k4MarlinWrapper/src/components/MarlinProcessorWrapper.cpp:112]: right, and then clearing the vector at the end of the loop ain't needed.

**Comment by andresailer** [comment]
This also needs an update https://github.com/key4hep/key4hep-doc/blob/master/examples/clic.md

**Comment by fdplacido** [comment]
> This also needs an update https://github.com/key4hep/key4hep-doc/blob/master/examples/clic.md

Done: https://github.com/key4hep/key4hep-doc/pull/25

**Review by andresailer — APPROVED** [review]
Much nicer than the END_TAG 👍

**Pull request merged**